File: Lecture_1_C/problem_set/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // 提示用户输入在1~8范围内的height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // 打印
    for (int i = 1; i <= height; i++)
    {
        for (int j = height - i; j > 0; j--)
        {
            printf(" ");
        }
        for (int j = i; j > 0; j--)
        {
            printf("#");
        }
        printf("  ");
        for (int j = i; j > 0; j--)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: Lecture_6_Python/problem_set/sentimental-mario-more/mario.py
from cs50 import get_int

num = get_int("Height: ")
while num < 1 or num > 8:
    num = get_int("Height: ")

for i in range(num):
    for j in range(num - i - 1):
        print(" ", end="")
    for j in range(i + 1):
        print("#", end="")
    print("  ", end="")
    for j in range(i + 1):
        print("#", end="")
    print()
